From: uazo <uazo@users.noreply.github.com>
Date: Sat, 29 Aug 2026 08:03:47 +0000
Subject: Allows v8 to be build in non-debug mode while Chromium is in debug
 mode

Allows V8 to be built in non-debug mode even in debug builds of Chromium,
as V8 is unusable in this mode within the emulator.

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 v8/BUILD.gn                                      | 6 ++++++
 v8/src/snapshot/embedded/embedded-file-writer.cc | 2 ++
 v8/src/snapshot/embedded/embedded-file-writer.h  | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/v8/BUILD.gn b/v8/BUILD.gn
--- a/v8/BUILD.gn
+++ b/v8/BUILD.gn
@@ -8069,6 +8069,12 @@ if (current_toolchain == v8_snapshot_toolchain) {
       ":disable_icf",
     ]
 
+    if (is_debug && !v8_enable_debugging_features) {
+      defines = [
+        "V8_SKIP_DEBUGGING_FEATURES_VALUE",
+      ]
+    }
+
     deps = [
       ":v8_base_without_compiler",
       ":v8_compiler_for_mksnapshot",
diff --git a/v8/src/snapshot/embedded/embedded-file-writer.cc b/v8/src/snapshot/embedded/embedded-file-writer.cc
--- a/v8/src/snapshot/embedded/embedded-file-writer.cc
+++ b/v8/src/snapshot/embedded/embedded-file-writer.cc
@@ -88,8 +88,10 @@ void EmbeddedFileWriter::WriteBuiltin(PlatformEmbeddedFileWriterBase* w,
   v8::internal::SourcePositionTableIterator positions(
       vpos, SourcePositionTableIterator::kExternalOnly);
 
+#ifndef V8_SKIP_DEBUGGING_FEATURES_VALUE
 #ifndef DEBUG
   CHECK(positions.done());  // Release builds must not contain debug infos.
+#endif
 #endif
 
   // Some builtins (InterpreterPushArgsThenFastConstructFunction,
diff --git a/v8/src/snapshot/embedded/embedded-file-writer.h b/v8/src/snapshot/embedded/embedded-file-writer.h
--- a/v8/src/snapshot/embedded/embedded-file-writer.h
+++ b/v8/src/snapshot/embedded/embedded-file-writer.h
@@ -103,9 +103,11 @@ class EmbeddedFileWriter : public EmbeddedFileWriterInterface {
   }
 
   void WriteExternalFilenames(PlatformEmbeddedFileWriterBase* w) const {
+#ifndef V8_SKIP_DEBUGGING_FEATURES_VALUE
 #ifndef DEBUG
     // Release builds must not contain debug infos.
     CHECK_EQ(external_filenames_by_index_.size(), 0);
+#endif
 #endif
 
     w->Comment(
--
